src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -302,6 +302,10 @@ void BaseCouplingScheme::initialize()
       // reserve memory and initialize data with zero
       if (_acceleration) {
         _acceleration->initialize(getAccelerationData());
+        std::shared_ptr<precice::acceleration::BaseQNAcceleration> qnAcceleration = std::dynamic_pointer_cast<precice::acceleration::BaseQNAcceleration>(_acceleration);
+        if (qnAcceleration) {
```

Comment:
You can use auto and the if init statement to make this a bit more compact. 

```suggestion
        if (auto qnAcceleration = std::dynamic_pointer_cast<precice::acceleration::BaseQNAcceleration>(_acceleration);
            qnAcceleration) {
```